mulog-kinesis/test/com/brunobonacci/mulog/publishers/test_publisher.clj

```diff
@@ -0,0 +1,56 @@
+(ns com.brunobonacci.mulog.publishers.test-publisher
+  (:require [com.brunobonacci.mulog :as μ]
+            [cheshire.core :as json]
+            [cognitect.aws.client.api :as aws])
+  (:import (java.util.concurrent TimeUnit)))
+
+(def KINESIS-LOCAL-STREAM-NAME "mulog-test-stream")
+
+(def KINESIS-LOCAL-SETTINGS {:api :kinesis
+                             :endpoint-override
+                                  {:protocol :http
+                                   :hostname "localhost"
+                                   :port 4568}})
+
+(defmacro with-local-kinesis-client
+  []
+  `(aws/client KINESIS-LOCAL-SETTINGS))
+
+(defmacro with-local-kinesis-stream
+  [client op params]
+  `(let [rq# (merge ~params {:StreamName KINESIS-LOCAL-STREAM-NAME})]
+     (aws/invoke ~client {:op ~op :request rq#})))
+
+(defmacro with-local-kinesis-publisher
+  [command]
+  `(let [sp# (μ/start-publisher!
+                  {:type                  :kinesis
+                   :stream-name           KINESIS-LOCAL-STREAM-NAME
+                   :kinesis-client-params KINESIS-LOCAL-SETTINGS})]
+    (do
+      (~@command)
+      ;; delay for kinesis processing
+      (.sleep (TimeUnit/SECONDS) 5)
+      (sp#))))
```

Comment:
maybe a better approach would be to the macro `with-local-kinesis-publisher` to work as follow:

```
(with-local-kinesis-publisher
   (µ/log :one-event :key1 "value1"))
;;= > [ {:mulog/event-name :one-event :key "value1"}]
```

in addition, we could have a
the `with-test-stream` should do the following operations:

1. create a stream with a random name like "mulog-test-234567889765"
2. wait for the stream to become available
3. run the `command` 
4. read the events from the stream
5. destroy the stream
6. return the events read in step 4

so that the test could look like

```
(with-test-stream
  (with-local-kinesis-publisher
     (µ/log :one-event :key1 "value1")))
;;= > [ {:mulog/event-name :one-event :key "value1"}]
```